src/OpenTelemetry.Extensions.Hosting/AssemblyInfo.cs

```diff
@@ -19,6 +19,7 @@
 [assembly: InternalsVisibleTo("OpenTelemetry.Extensions.Hosting.Tests" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("DynamicProxyGenAssembly2" + AssemblyInfo.MoqPublicKey)]
 
+#if EXPOSE_EXPERIMENTAL_FEATURES
```

Comment:
Why is this being wrapped in the experimental check?

Reply:
When `!EXPOSE_EXPERIMENTAL_FEATURES` than `OpenTelemetry.Extension.Hosting` [sees the internals of OpenTelemetry.Api](https://github.com/open-telemetry/opentelemetry-dotnet/blob/76d5a7c714fc9d4a5b3680e6f8993385a395f8c3/src/OpenTelemetry.Api/AssemblyInfo.cs#L32) and it also sees the [internal static class AssemblyInfo](https://github.com/open-telemetry/opentelemetry-dotnet/blob/76d5a7c714fc9d4a5b3680e6f8993385a395f8c3/src/OpenTelemetry.Api/AssemblyInfo.cs#L37) from that same file.

It is essentially yet another `InternalsVisibleTo` problem 🤣 

Check out [this commit](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4735/commits/88d3273c6ad3d3dbac231a35142fad1cdd6b3260) I pushed. I had the idea to use this new [C# 11 file-local types](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/file-local-types) feature to prevent these `AssemblyInfo` helper classes from flowing to other assemblies. Let me know what you think about that!